tmgmt_zanata.connector.inc

```diff
@@ -362,9 +367,230 @@ class TmgmtZanataConnector {
+                array('@error' => $response->error)),
+              'error',
+              FALSE);
+            return FALSE;
+
+          case 404:
+            $this->showDocumentNotFoundErrorMessage();
+            return FALSE;
+
+          case 401:
+            $this->showAuthErrorMessage();
+            return FALSE;
+
+          case 500:
+          default:
+            drupal_set_message(
+              t('There was a problem sending translations to Zanata. See messages below.'),
+              'warning',
+              FALSE);
+            $job_item->addMessage(
+              'Tried to send existing translations but got response code @code when sending to URL @url.',
+              array(
+                '@code' => $response->code,
+                '@url' => $url));
+            return FALSE;
+        }
+      }
+      else {
+        // no existing translations found. Probably don't need a log message.
+      }
+
+    }
+    else {
+      $job_item->addMessage(
+        'Item "@id" is not type "node" (is type "@type"), unable to look up translations.',
+        array(
+          '@id' => $id,
+          '@type' => $job_item->item_type,
+        ));
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * $data should be an already filtered and flattened array of translations
+   *       that all have an id matching a source string id.
+   */
+  private function prepareTargets($data, $job_item) {
+    if ($this->useSegmentation) {
+      // Not completely reliable if translations do not follow same sentence or
+      // paragraph structure (segments may not line up properly between source
+      // and translation).
+      $source_data = $this->getFlattenedSourceData($job_item);
+      $data = $this->segmenter->segmentTranslations($source_data, $data);
+    }
+
+    $items = array();
+    $document = $this->getDocId($job_item);
+    foreach ($data as $key => $value) {
+      $text = $value['#text'];
+      $items[] = $this->createTranslationItem($key, $text, $document);
+    }
+
+    return $items;
+  }
+
+  /**
+   * Create an associative array for a translation item ready for JSON
+   * encoding. The format is suitable for the Zanata server.
+   *
+   * This is for sending existing translations to Zanata.
+   */
+  private function createTranslationItem($id, $text, $document) {
+    return array(
+      'resId' => $id,
+      'state' => 'NeedReview',
+      'content' => $text,
+      'extensions' => array(
+        array(
+          "object-type" => "comment",
+          "space" => "preserve",
+          "value" => "Translation sent by tmgmt-zanata because it matches id \"$id\" in document \"$document\"."
+        ),
+      ),
+      'revision' => 1,
+      'textFlowRevision' => 1,
+    );
+  }
+
+  private function getExistingTransData(TMGMTJobItem $job_item) {
+    $id = $job_item->item_id;
+    $node = node_load($id);
+
+    $translations = translation_node_get_translations($node->tnid);
+
+    if ($translations) {
```

Comment:
- [ ] Change translation check else portion to a pre-check that returns if failed.